core/domain/training_job_result_domain_test.py

```diff
@@ -0,0 +1,42 @@
+# coding: utf-8
+#
+# Copyright 2020 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS-IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Tests for gce metadata services"""
+
+from core.domain import training_job_result_domain
+from core.domain.protofiles import text_classifier_pb2
+from core.tests import test_utils
+
+class TrainingJobResultTests(test_utils.GenericTestBase):
+    """Tests for TrainingJobResult domain object."""
+
+    def test_that_validation_checks_are_correct(self):
```

Comment:
What does ``correct`` mean? When are they? It seems from the test body that correct means an exception is sometimes thrown, and other times not. In general, it's a good practice for each test to test a unique behavior, and to be as specific as to what that behavior is. For example, you might be able to split this up as so:

```python
def test_validate_for_text_classifier_with_valid_model_does_not_raise_exception()
  ...

def test_validate_for_text_classifier_with_invalid_model_raises_exception()
  ...
```

Also note that this style of test naming can help you discover other cases that might be worth testing (e.g. maybe another classifier type, or an invalid classifier type, or maybe we should be more specific about what designates a valid/invalid model).

Ditto for your other tests below.

Reply:
The validation is really simple, it checks that the `classifier_data` stores the data of the classifier corresponding to the `algorithm_id` field. Hence, other types of the classifier are not necessary and this also defines what a valid job data is and an invalid job data is. I will add a comment in the validation check function to explain this clearly. That should make it clear the valid / invalid job data.

Reply:
Done.